sympy/functions/special/bessel.py

```diff
@@ -234,10 +234,15 @@ def _eval_nseries(self, x, n, logx, cdir=0):
         from sympy.series.order import Order
         nu, z = self.args
 
+        # In case of powers less than 1, number of terms need to be computed
+        # separately to avoid repeated callings of _eval_nseries with wrong n
+        _, exp = z.as_leading_term(x).as_base_exp()
+        newn = n//exp + 1
```

Comment:
Does this work also when `exp` is not an integer?

Reply:
I would try something like `ceiling(n/exp)`.

Reply:
Yes, I believe it is working for float `exp`. Here is the output (with `print(exp, newn)`) - 
```py
In [3]: besselj(1, x).series(x)
1 7
Out[3]: 
     3     5        
x   x     x     ⎛ 6⎞
─ - ── + ─── + O⎝x ⎠
2   16   384        

In [4]: besselj(1, x**(2.5)).series(x)
2.50000000000000 3
Out[4]: 
 2.5        
x       ⎛ 6⎞
──── + O⎝x ⎠
 2          

In [5]: besselj(1, x**(0.5)).series(x)
0.500000000000000 13
Out[5]: 
      5.5        4.5      3.5    2.5    1.5    0.5        
     x          x        x      x      x      x       ⎛ 6⎞
- ───────── + ─────── - ───── + ──── - ──── + ──── + O⎝x ⎠
  176947200   1474560   18432   384     16     2          
```

Reply:
newn should also be an integer.

Reply:
But newn is indeed an integer. `n//exp` is floor division and I am adding an integer (1) to it. I think ceiling(n/exp) is probably a better way to write this line, so will make the change